src/platforms/binary32.rkt

```diff
@@ -46,35 +46,134 @@
   (begin
     (define-libm-impls/binary32* (itype ... otype) name ...) ...))
 
-(define-operator-impl (neg.f32 [x : binary32])
-                      binary32
-                      #:spec (neg x)
-                      #:fpcore (! :precision binary32 (- x))
-                      #:fl fl32-)
+(define-operator-impl
+ (neg.f32 [x : binary32])
+ binary32
+ #:spec (neg x)
+ #:fpcore (! :precision binary32 (- x))
+ #:fl fl32-
+ #:identities (#:exact (neg.f32 x)
+                       [distribute-lft-neg-in (neg.f32 (*.f32 a b)) (*.f32 (neg.f32 a) b)]
+                       [distribute-rgt-neg-in (neg.f32 (*.f32 a b)) (*.f32 a (neg.f32 b))]
+                       [distribute-neg-in (neg.f32 (+.f32 a b)) (+.f32 (neg.f32 a) (neg.f32 b))]
+                       [distribute-neg-frac (neg.f32 (/.f32 a b)) (/.f32 (neg.f32 a) b)]
+                       [distribute-neg-frac2 (neg.f32 (/.f32 a b)) (/.f32 a (neg.f32 b))]
+                       [remove-double-neg (neg.f32 (neg.f32 a)) a]
+                       [neg-sub0 (neg.f32 b) (-.f32 0 b)]
+                       [neg-mul-1 (neg.f32 a) (*.f32 -1 a)]))
 
-(define-operator-impl (+.f32 [x : binary32] [y : binary32])
-                      binary32
-                      #:spec (+ x y)
-                      #:fpcore (! :precision binary32 (+ x y))
-                      #:fl fl32+)
+(define-operator-impl
+ (+.f32 [x : binary32] [y : binary32])
+ binary32
+ #:spec (+ x y)
+ #:fpcore (! :precision binary32 (+ x y))
+ #:fl fl32+
+ #:identities (#:commutes [distribute-neg-out (+.f32 (neg.f32 a) (neg.f32 b)) (neg.f32 (+.f32 a b))]
+                          [+-lft-identity (+.f32 0 a) a]
+                          [+-rgt-identity (+.f32 a 0) a]
+                          [unsub-neg (+.f32 a (neg.f32 b)) (-.f32 a b)]))
 
 (define-operator-impl (-.f32 [x : binary32] [y : binary32])
                       binary32
                       #:spec (- x y)
                       #:fpcore (! :precision binary32 (- x y))
-                      #:fl fl32-)
+                      #:fl fl32-
+                      #:identities
+                      ([cancel-sign-sub (-.f32 a (*.f32 (neg.f32 b) c)) (+.f32 a (*.f32 b c))]
+                       [cancel-sign-sub-inv (-.f32 a (*.f32 b c)) (+.f32 a (*.f32 (neg.f32 b) c))]
+                       #:exact (-.f32 a a)
+                       #:exact (-.f32 a 0)
+                       [sub0-neg (-.f32 0 a) (neg.f32 a)]
+                       [sub-neg (-.f32 a b) (+.f32 a (neg.f32 b))]))
 
-(define-operator-impl (*.f32 [x : binary32] [y : binary32])
-                      binary32
-                      #:spec (* x y)
-                      #:fpcore (! :precision binary32 (* x y))
-                      #:fl fl32*)
+(define-operator-impl
+ (*.f32 [x : binary32] [y : binary32])
+ binary32
+ #:spec (* x y)
+ #:fpcore (! :precision binary32 (* x y))
+ #:fl fl32*
+ #:identities (#:commutes [distribute-lft-neg-out (*.f32 (neg.f32 x) y) (neg.f32 (*.f32 x y))]
```

Comment:
Should `#:commutes` be at the `define-operator-impl` level? Just a thought

Reply:
It should be